src/super_gradients/training/kd_model/kd_model.py

```diff
@@ -20,7 +20,7 @@ class KDModel(SgModel):
 
     def build_model(self,  # noqa: C901 - too complex
                     architecture: Union[str, nn.Module] = 'kd_module',
-                    arch_params={},
+                    arch_params={}, checkpoint_params={},
```

Comment:
@shaydeci  - You've worked really hard to move the checkpoint loading mechanism of sg_model outside of build_model, yet in KDModel you break this and add this unneeded coupling again.

Can we implement the same logic as we did in SGModel or is this something I'm missing?

Reply:
The logic isn't changed.
Load checkpoint was always called inside build_model, decoupling it (in SG or KD) is not something that was discussed when we got Algo's comments regarding KD implementation, but rather the fact that arch_params holds extra information which is not architecture related....

Reply:
This is correct... The problem is with the end result of we have right now on our plate:
1. Inside SGModel -> De-Coupled logic for "build_model"(A) and "load_checkpoint"(B) but B is invoked from A
2. Inside KDModel -> logic for "load_checkpoint"(B) is manipulated because it is called from within A... So it feels like a good place to "add just a little more business logic to not touch load_checkpoint in the inheriting class" when in fact we want the exact opposite... We want the inheriting class to make the proper adjustments inside "load_checkpoint" and than he can (if this is what's needed) to call super.load_checkpoint after running the new BL...